src/t8_cmesh/t8_cmesh_readmshfile.cxx

```diff
@@ -1775,12 +1841,8 @@ t8_cmesh_from_msh_file (const char *fileprefix, int partition,
         t8_msh_file_4_read_nodes (file, &num_vertices, &node_mempool);
       if (use_occ_geometry) {
 #if T8_WITH_OCC
-        geometry_occ = t8_geometry_occ_new (dim, fileprefix, "brep_geometry");
-        geometry = geometry_occ;
-        /* Register geometry */
-        t8_cmesh_register_geometry (cmesh, geometry);
         t8_cmesh_msh_file_4_read_eles (cmesh, file, vertices, &vertex_indices,
-                                       dim, geometry_occ);
+                                       dim, linear_geometry, 1, occ_geometry);
```

Comment:
Since the occ geometry will be set to the geometry or to NULL by `t8_cmesh_from_msh_file_register_geometries`, we can ditch the if statement and just use
```
t8_cmesh_msh_file_4_read_eles (cmesh, file, vertices, &vertex_indices,
                                       dim, linear_geometry, use_occ_geometry, occ_geometry);
```

Reply:
DONE